pmlc/dialect/abi/ir/ops.td

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Intel Corporation
+
+#ifndef __PML_ABI_OPS__
+#define __PML_ABI_OPS__
+
+include "mlir/Dialect/StandardOps/IR/StandardOpsBase.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+include "pmlc/util/enums.td"
+
+def ABI_Dialect : Dialect {
+  let name = "abi";
+  let cppNamespace = "pmlc::dialect::abi";
+}
+
+class ABI_Op<string mnemonic, list<OpTrait> traits = []> :
+    Op<ABI_Dialect, mnemonic, traits>;
+
+def ABI_LoopOp : ABI_Op<"loop", [
+  IsolatedFromAbove, HasParent<"ModuleOp">, RecursiveSideEffects]> {
+  let summary = "Repeat runs of a network";
+  let description = [{
+    The "abi.loop" operation represents repeated runs of a program,
+    taking new inputs and generating new outputs each time.
+    This allows loop-invariant code motion to move setup and teardown
+    operations outside of the loop, so that their cost is ammortized over
+    many runs of the program.
+  }];
+  let regions = (region AnyRegion:$initRegion, AnyRegion:$bodyRegion, AnyRegion:$finiRegion);
+  let arguments = (ins TypeArrayAttr:$networkFieldTypes);
+  let builders = [OpBuilder<"">];
+  let skipDefaultBuilders = 1;
+  let assemblyFormat = [{
+     `init` $initRegion `yielding` $networkFieldTypes `to` `body` $bodyRegion `and` `to` `fini` $finiRegion attr-dict
```

Comment:
I'd prefer that we use less verbiage and keep this more like assembly language where we have ops with symbols instead of using verbs in between parts.

I also think we should include block arguments in the custom syntax so that the default printing of block arguments is avoided.

Reply:
Verbiage: decreased.

I'm not sure how to include block arguments in the custom syntax -- got an example of where this is done?

Reply:
Checkout the AffineParallelOp for an example.